graph/generator.js

```diff
@@ -14,63 +16,36 @@ function transformArgument(arg) {
 
 function transformField(field) {
   return {
+    type: getBaseType(field.type).name,
+    kind: getBaseType(field.type).kind,
     fieldName: field.name,
-    isList: hasListType(field.type)
+    isList: hasListType(field.type),
+    args: (field.args || []).map(transformArgument)
   };
 }
 
 function objectifyField(acc, field) {
-  const descriptor = {};
-
-  Object.keys(field).filter(key => {
+  const descriptor = Object.keys(field).filter(key => {
     return (key !== 'fieldName');
-  }).forEach(key => {
-    descriptor[key] = field[key];
-  });
+  }).reduce((descriptorAcc, key) => {
+    descriptorAcc[key] = field[key];
+
+    return descriptorAcc;
+  }, {});
 
   acc[field.fieldName] = descriptor;
 
   return acc;
 }
 
-function transformFieldWithArgs(field) {
-  return {
-    fieldName: field.name,
-    isList: hasListType(field.type),
-    args: field.args.map(transformArgument)
-  };
-}
-
-function transformRelationship(field) {
-  let args;
-
-  if (field.args) {
-    args = field.args.map(transformArgument);
-  } else {
-    args = [];
-  }
-
-  const type = getBaseType(field.type);
-  const isList = hasListType(field.type);
-
-  const relationship = {
-    fieldName: field.name,
-    isList,
-    type,
-    args
-  };
-
-  return relationship;
-}
-
 function getParents(typeName, typeList) {
   return typeList.filter(potentialParent => {
     if (!potentialParent.fields) {
       return false;
     }
 
     return (potentialParent.fields.filter(field => {
-      return getBaseType(field.type) === typeName;
+      return getBaseType(field.type).name === typeName;
     }).length > 0);
```

Comment:
What about `some` instead of `filter` + length check? (if that function is available in this build)